frontend/dockerfile/builder/build.go

```diff
@@ -767,6 +769,79 @@ func scopeToSubDir(c *llb.State, fileop bool, dir string) *llb.State {
 	return &bc
 }
 
+func contextByName(c client.Client, p *ocispecs.Platform) func(context.Context, string) (*llb.State, *dockerfile2llb.Image, error) {
+	return func(ctx context.Context, name string) (*llb.State, *dockerfile2llb.Image, error) {
+		named, err := reference.ParseNormalizedNamed(name)
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "invalid context name %s", name)
+		}
+		name = strings.TrimSuffix(reference.FamiliarString(named), ":latest")
+
+		opts := c.BuildOpts().Opts
+		v, ok := opts["context:"+name]
+		if !ok {
+			return nil, nil, nil
+		}
+
```

Comment:
Can we prohibit overriding predefined contexts such as `context:scratch` ?

Reply:
Agree. I think we should also block `context:context` and reserve that key to the "main build context".

Reply:
Added